AK/HashTable.h

```diff
@@ -378,31 +396,36 @@ class HashTable {
         auto old_capacity = m_capacity;
         Iterator old_iter = begin();
 
-        if constexpr (IsOrdered) {
-            m_buckets = (BucketType*)kmalloc(size_in_bytes(new_capacity));
-            __builtin_memset(m_buckets, 0, size_in_bytes(new_capacity));
+        auto new_buckets = kmalloc(size_in_bytes(new_capacity));
+        if (!new_buckets)
+            return false;
```

Comment:
The change to how `m_collection_data` is assigned is a change that shouldn't have been bundled into this commit, as it's hard to verify the diff now. Please move this to another commit.

Reply:
This does not change how it is assigned, but only change the duplicate allocation lines into only one that honors allocation failure.
The other line was just moved down to cut down in if statements